### DISCUSSION TITLE
Having trouble with Sunsynk Inverter Add-on (multi)

### DISCUSSION THREAD

**Discussion by Seclar** [kellerza/sunsynk#108] [Q&A] [answered]
Hi

I have two Sunsynk 5kW Hybrid Inverters. At the moment I have a USR-TCP232-304 Serial RS485 to TCP/IP Ethernet Server Converter Module plugged into each one and I am successfully reading data from one using the non-multi version of the Sunsynk Inverter Add-on. 

I wanted to try using this add-on to read data from both inverters. I started off by trying to read from just one but I can't even save config - it throws an error. Below is the config I've used. 

DRIVER: umodbus
INVERTERS:
  - SERIAL_NR: "2299999999"
    MODBUS_ID: 1
    HA_PREFIX: "sunsynkFR"
    PORT: "tcp://192.168.50.242:502"
    DEVICE: "/dev/ttyS2"

SENSORS:
  - total_pv_energy
  - total_battery_charge
  - total_battery_discharge
  - total_grid_export
  - total_grid_import
  - battery_soc

MQTT_HOST: core-mosquitto
MQTT_PORT: 1883
MQTT_USERNAME: mozzy
MQTT_PASSWORD: nopassword
DEBUG: 2
MANUFACTURER: Sunsynk
READ_SENSORS_BATCH_SIZE: 60
NUMBER_ENTITY_MODE: auto

I cannot, no matter what I try, get it to save the config. It keeps throwing this syntax error:

"Failed to save add-on configuration, invalid value or type (must have a partial ordering). Got {'DRIVER': 'umodbus', 'INVERTERS': [{'SERIAL_NR':..."

As far as I can tell that message comes from a Python library that is syntax checking the name-value pairs. I've gone through the documentation and checked. All entries I have appear to be valid syntax and I've tried removing quotes, adding quotes etc but to no avail. Because I can't save the config I can't even start the module to try it out.

Any ideas?

Paul

**Comment by Seclar** [comment, marked as answer]
Have I missed something in the above configuration? 

Has anyone got his multi version of the mod working? If so, would you mind posting your config?

**Reply by kellerza** [reply]
You should not be using it yet

**Reply by kellerza** [reply]
You can now try the multi addon :wink: https://kellerza.github.io/sunsynk/reference/multi-options
